one/src/cbor_value.rs

```diff
@@ -0,0 +1,285 @@
+    Bool(bool),
+    /// Simple Values (22 null)
+    /// https://datatracker.ietf.org/doc/html/rfc8949#fpnoconttbl2
+    Null,
+    /// Simple Values (23 undefined)
+    /// https://datatracker.ietf.org/doc/html/rfc8949#fpnoconttbl2
+    Undefined,
+    /// Major type 0: An unsigned integer in the range 0..(2**64)-1 inclusive.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    U64(u64),
+    /// Major type 1: A negative integer in the range -2**64..-1 inclusive.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    /// note: we store Major type 1: A negative integer less then -2**63 in a i128
+    I64(i64),
+    /// Major type 1: A negative integer in the range -2**64..-1 inclusive.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    /// note: we store Major type 1: A negative integer grater then -2**63 in a i64
+    Int(i128), // Cbor ints range from [ -(2^64) , 2^64-1 ] to fit this i65 we use i128
+    /// Major type 7:
+    /// Floating-point numbers and simple values, as well as the "break" stop code.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#fpnocont
+    /// note: we store Simple values and breaks separately
+    F64(OrderedFloat<f64>), // this makes it possible to put CborValue in BTreeMap
+    /// Simple Values (20 false | 21 true)
+    /// https://datatracker.ietf.org/doc/html/rfc8949#fpnoconttbl2
+    /// 24..31 (reserved)
+    /// 32..255 (unassigned)
+    Simple(u8),
+    /// Major type 2: A byte string.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    Bytes(Vec<u8>),
+    /// Major type 3: A text string
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    String(String),
+    /// Major type 4: An array of data items.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    Array(Vec<CborValue>),
+    /// Major type 5: A map of pairs of data items.
+    /// https://datatracker.ietf.org/doc/html/rfc8949#section-3.1
+    Map(BTreeMap<CborValue, CborValue>),
+    /// Major type 6:
+    /// A tagged data item ("tag") whose tag number, an integer in the range 0..(2**64)-1 inclusive
+    /// https://datatracker.ietf.org/doc/html/rfc8949#tags
+    /// https://www.iana.org/assignments/cbor-tags/cbor-tags.xhtml
+    /// tag(42) = Cid
+    Tag((Tag, Box<CborValue>)),
+    /// Major type 7: 0xff Break signal
+    /// https://datatracker.ietf.org/doc/html/rfc8949#break
+    Break, // 0xff
+    /// Used to represent a byte that was not valid for its location.
+    /// You should not see this from a well formed CBOR value.
+    Unknown(u8),
+}
+
+impl CborValue {
+    /// Parse bytes to a single CborValue
+    /// This will error on CBOR seq.
+    pub fn parse(bytes: &[u8]) -> Result<Self> {
+        let mut decoder = Decoder::new(bytes);
+        let c = CborValue::next(&mut decoder);
+        if decoder.position() != bytes.len() {
+            debug!("decoder not at end {}/{}", decoder.position(), bytes.len());
```

Comment:
should this be a warn (or at least info?) i guess we sort of have a "if you log it, you handled it rule" but it'd be nice if the error returned had a bit more info about why it happened (does this mean you somehow packed multiple cbor values into a single thing, or it was just bad input?).

Reply:
I disagree asking to decode a thing and it not being valid is a normal thing that can happen. I would prefer not to log at all unless the caller has a reson to beleve that a failed parse is suspicions. 

What info would be helpfull for the Err?
The two main resons for a not at end error is ether it was `CBOR seq` a sequence of CBOR objects using the fact that CBOR knows its length to not need framing or it is just not a CBOR object at all so it is obviously not a formed CBOR object.

Reply:
Yeah, that's fair. I guess I wasn't sure if this was internal parsing and it should only fail if we did something wrong, in which case it's nice to know and have all the info about what, or if it's validating any data in which was we expect to fail regularly. I don't know what would be helpful, it's fine to close.